### ISSUE TITLE
can't use `:srcObject.prop` to set prop to null

### ISSUE THREAD

**Issue by freddialpad** [vuejs/vue#9138]
### Version
2.5.17

### Reproduction link
[https://codesandbox.io/s/ppyq71wrxj](https://codesandbox.io/s/ppyq71wrxj)

### Steps to reproduce
(See minimal repro)

1. have a component with an `HTMLMediaElement` that binds `:srcObject.prop` to some data property (that starts off as `null`):
```
<audio :srcObject.prop="stream">
```
2. set it to some valid `MediaStream`:
```
this.stream = new MediaStream();
```
3. set it back to `null`:
```
this.stream = null;
```

### What is expected?
I expect the audio.srcObject to be null

### What is actually happening?
it stays as the MediaStream and there's a TypeError in the log.

---
I noticed that it did actually work in my first attempt to write the repro example: https://jsfiddle.net/4tnkapxo/1/ (probably because the whole thing is inline?)

Im not sure how helpful this is, but based on the stack trace it's related to this code, found in `updateDOMProps`:

```
  for (key in oldProps) {
    if (isUndef(props[key])) {
      elm[key] = '';
    }
  }
```

`isUndef(null)` is evaluating true, and so it's trying to set the prop to `''`, but `srcObject` specifically needs it to be `null` or a `MediaStream`

**Comment by posva** [comment]
maybe we shouldn't apply transformation for `.prop` bindings

**Comment by superfreddialpad** [comment]
I think this was fixed by f11449d916a468651d4fd5024c37e3eebbc9941f
